clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -184,13 +184,83 @@ class AzureNetworkClient extends AzureBaseClient {
     def loadBalancer = client.getLoadBalancersOperations().get(resourceGroupName, loadBalancerName, null).body
 
     if (loadBalancer.frontendIPConfigurations.size() != 1) {
-      throw new RuntimeException("Unexpected number of public IP addresses associated with the load balancer (should be only one)!")
+      throw new RuntimeException("Unexpected number of public IP addresses associated with the load balancer (should always be only one)!")
     }
 
     def publicIpAddressName = AzureUtilities.getResourceNameFromID(loadBalancer.frontendIPConfigurations.first().getPublicIPAddress().id)
-    client.getLoadBalancersOperations().delete(resourceGroupName, loadBalancerName)
 
-    client.getPublicIPAddressesOperations().delete(resourceGroupName, publicIpAddressName)
+    ServiceResponse<Void> result = null
+    // The API call might return a timeout exception or some other Azure CloudException that is not
+    //   related to the operation we are trying to execute; retry couple of times and if the final
+    //   retry fails then rethrow
+    long operationRetry = 0
+    while (operationRetry < AZURE_ATOMICOPERATION_RETRY) {
```

Comment:
Looks like this behavior is used in a handful of places, I'd suggest a method (taking a closure) that wraps the retry behavior and just calls out to the closure to actually execute the client call.

This is primarily towards eliminating copy & pasted code.